src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectEngineTracker.cs

```diff
@@ -55,12 +59,33 @@ public RazorProjectEngine GetProjectEngine(ProjectSnapshot snapshot)
                     if (_projectEngine == null)
                     {
                         var factory = _services.GetRequiredService<ProjectSnapshotProjectEngineFactory>();
-                        _projectEngine = factory.Create(snapshot);
+                        _projectEngine = factory.Create(
+                            state.HostProject.Configuration, 
+                            RazorProjectFileSystem.Create(Path.GetDirectoryName(state.HostProject.FilePath)), 
+                            configure: null);
                     }
                 }
             }
 
             return _projectEngine;
         }
+
+        public List<string> GetImportDocumentTargetPaths(ProjectState state, string targetPath)
+        {
+            var projectEngine = GetProjectEngine(state);
+            var importFeature = projectEngine.ProjectFeatures.OfType<IImportProjectFeature>().FirstOrDefault();
+            var projectItem = projectEngine.FileSystem.GetItem(targetPath);
+            var importItems = importFeature?.GetImports(projectItem).Where(i => i.FilePath != null);
+
+            // Target path looks like `Foo\\Bar.cshtml`
+            var targetPaths = new List<string>();
+            foreach (var importItem in importItems)
+            {
+                var itemTargetPath = importItem.FilePath.Replace('/', '\\').TrimStart('\\');
```

Comment:
Why normalize to `\\` instead of `Path.DirectorySeparator`?

Reply:
because it seemed straightforward? By convention paths in msbuild are specified windows styel

Reply:
I'd verify this returns what you'd expect cross-plat. The Razor's file system abstraction interacts directly with the file system.

Reply:
These aren't used by the file system. These are keys. Target path determines what imports apply to what document, it doesn't determine where those documents are located.